andela/ah-backend-prime
# 164798296 Users can like/dislike an article

**What does this PR do?**
    Creates functionality for users to like or dislike an article

**Description of tasks to be completed**
   

- Create a view to handle liking and disliking an article

- Create LikeDislike model

- Create tests for like-dislike functionality

**How should this be manually tested?**

- After cloning the repo, cd into it and run the following commands to activate the virtual environment
   `virtualenv venv`
   `source venv/bin/activate`

- Checkout the branch with the likes-dislikes functionality
   `git checkout ft-likes-and-dislikes-164798296`

- Install all requirements
   `pip install -r requirements.txt`

- Source the environment variables
   `source .env`

- Run the migrations
   `python manage.py makemigrations`
   `python manage.py migrate`

- Run the server
   `python manage.py runserver`

- To test out this functionality one has to first register on the application, in case of a successful
   registration, they will receive an account activation link on their email, this activation link would then
   be pasted in the URL section of postman and sent in order to activate and log into one's account
   where they would get a token which they would use in the Authorization section to access all other
   routes including the tags route.

- To like an article, you have to get the slug(unique article identifier) of the particular article you want to like and pass it in the **LIKES** url: `http://127.0.0.1:8000/api/v1/articles/{article-slug}/like/`.  (See Figure 1)

<img width="1219" alt="Screen Shot 2019-04-15 at 15 17 33" src="https://user-images.githubusercontent.com/26630001/56138829-53645080-5fa0-11e9-997a-930562a7d504.png">

**Figure 1**

- To dislike an article, you have to get the slug(unique article identifier) of the particular article you want to dislike and pass it in the **DISLIKES** url: `http://127.0.0.1:8000/api/v1/articles/{article-slug}/dislike/`.  (See Figure 2)

<img width="1215" alt="Screen Shot 2019-04-15 at 15 17 58" src="https://user-images.githubusercontent.com/26630001/56138995-a3dbae00-5fa0-11e9-9c67-aff5d3eccb95.png">

**Figure 2**


**What are the relevant pivotal tracker stories?**
    [#164798296](https://www.pivotaltracker.com/story/show/164798296)
